test/2fa.test.js

```diff
@@ -0,0 +1,263 @@
+});
+
+afterEach(() => {
+    console.log = ctx.savedLog;
+});
+
+const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+
+const twoFactorMethods = [
+    { kind: '2fa-email', detail: 'hello@example.com', publicKey: 'pkemail' },
+    { kind: '2fa-phone', detail: '+1 717 555 0101', publicKey: 'pkphone' },
+];
+
+const inMemorySigner = new nearAPI.InMemorySigner(keyStore);
+async function signatureFor(accountId, valid = true) {
+    let blockNumber = (await ctx.near.connection.provider.status()).sync_info.latest_block_height;
+    if (!valid) blockNumber = blockNumber - 101;
+    blockNumber = String(blockNumber);
+    const message = Buffer.from(blockNumber);
+    const signedHash = await inMemorySigner.signMessage(message, accountId);
+    const blockNumberSignature = Buffer.from(signedHash.signature).toString('base64');
+    return { blockNumber, blockNumberSignature };
+}
+
+const { parseSeedPhrase } = require('near-seed-phrase');
+const SEED_PHRASE = 'shoot island position soft burden budget tooth cruel issue economy destroy above';
+const keyPair = nearAPI.KeyPair.fromString(parseSeedPhrase(SEED_PHRASE).secretKey);
+async function createNearAccount(accountId) {
+    if (!accountId) {
+        accountId = `helper-test-${Date.now()}`;
+    }
+    const response = await request.post('/account')
+        .send({
+            newAccountId: accountId,
+            newAccountPublicKey: keyPair.publicKey.toString()
+        });
+    await keyStore.setKey(undefined, accountId, keyPair);
+    assert.equal(response.status, 200);
+    return accountId;
+}
+
+describe('setting up 2fa method', () => {
+    let accountId = 'testing' + Date.now();
+    let method = twoFactorMethods[0];
+    let securityCode = '';
+    let requestId = -1;
+    let testing = true;
+
+    test('generate a deterministic public key for an account', async () => {
+        await createNearAccount(accountId);
+
+        await request.post('/2fa/getAccessKey')
+            .send({
+                accountId,
+                ...(await signatureFor(accountId))
+            })
+            .expect('Content-Type', /json/)
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+                assert(res.body.publicKey.length > 80);
+            })
+            .expect(200);
+    });
+
+    test('initCode for an account, sets up 2fa method', async () => {
+
+        await request.post('/2fa/init')
+            .send({
+                testing,
+                accountId,
+                method,
+                ...(await signatureFor(accountId))
+            })
+            .expect('Content-Type', /json/)
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+                assert.equal(res.body.securityCode.length, 6);
+                securityCode = res.body.securityCode;
+            })
+            .expect(200);
+    });
+
+    test('should be able to call init again with different method', async () => {
```

Comment:
individual tests shouldn't depend on previous test success generally, otherwise it's hard to do one of them in isolation, execute tests in parallel, etc

Reply:
realized other tests were already doing it, so I guess just nevermind

Reply:
I'll see what I can do to evolve the 2fa testing and make it more isolated